src/components/Channel/Channel.tsx

```diff
@@ -475,6 +476,192 @@ const ChannelInner = <
     }
   }, [state.messages, state.thread]);
 
+  /** Keyboard Navigation */
+
+  const regularMessages = channel.state.messages.filter((m) => m.type === 'regular');
+  const numberOfRegularMessages = regularMessages.length;
+  const messageElements = document.getElementsByClassName('str-chat__message--regular');
```

Comment:
Relying on `document.getElementsByClassName` to reference DOM elements is quite the brittle approach for two reasons. 

1. It searches the global DOM. If for some reason, the document contains more than one message list, it will get them.
2. It also couples the implementation logic to styling classes. If those classes are changed for some reason, this code will stop working. 

The best way to avoid that is to use `refs`. If you need to iterate through multiple elements, then you can still query the DOM, but make that through `qerySelectorAll` of a **known parent** (which you reference with a `ref`) and use a data attribute for that (e.g. `data-role`).